### ISSUE TITLE
System.ArgumentOutOfRangeException

### ISSUE THREAD

**Issue by VitoDeLore** [PowerShell/PSReadLine#1112]
Ultimi 200 tasti:
 s s = l o c a l h o s t : 5 2 1 7 0 ' Space ' - D f i l e . e n c o d i n g = U T F - 8 ' Space ' - c p ' Space ' C : \ U s e r s \ v i t o d \ A p p D a t a \ R o a m i n g \ C o d e \ U s e r \ w o r k s p a c e S t o r a g e \ f 6 1 d a b 9 9 e d 0 b 3 f f 1 8 b f f f a 9 6 2 5 4 9 b a e 0 \ r e d h a t . j a v a \ j d t _ w s \ j d t . l s - j a v a - p r o j e c t \ b i n ' Space ' H e l l o W o r l d ' Space Enter        

Eccezione:
System.ArgumentOutOfRangeException: Il valore deve essere maggiore o uguale a zero e minore della grandezza del buffer della console in tale dimensione.
Nome parametro: top
Valore effettivo: -2.
   in System.Console.SetCursorPosition(Int32 left, Int32 top)
   in Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
address=localhost:52170' '-Dfile.encoding=UTF-8' '-cp' 'C:\Users\vitod\AppData\Roaming\Code\User\workspaceStorage\f61dab99ed0b3ff18bfffa962549bae0\redhat.java\jdt_ws\jdt.ls-java-project\bin' 'HelloWorld'
Hello

**Comment by Reelix** [comment]
OS: Windows 10 1903 Build 18362.418
Powershell Version: 5.1.18362.145
Windows Terminal Version: 0.5.2762.0

I got almost an almost identical error - Repeatable with these 3 steps.

1.) Open the new Windows Terminal
2.) Open Powershell through it
3.) Copy the following 26,425 characters

> aaaaaaaabaaaaaaacaaaaaaadaaaaaaaeaaaaaaafaaaaaaagaaaaaaahaaaaaaaiaaaaaaajaaaaaaakaaaaaaalaaaaaaamaaaaaaanaaaaaaaoaaaaaaapaaaaaaaqaaaaaaaraaaaaaasaaaaaaataaaaaaauaaaaaaavaaaaaaawaaaaaaaxaaaaaaayaaaaaaazaaaaaabbaaaaaabcaaaaaabdaaaaaabeaaaaaabfaaaaaabgaaaaaabhaaaaaabiaaaaaabjaaaaaabkaaaaaablaaaaaabmaaaaaabnaaaaaaboaaaaaabpaaaaaabqaaaaaabraaaaaabsaaaaaabtaaaaaabuaaaaaabvaaaaaabwaaaaaabxaaaaaabyaaaaaabzaaaaaacbaaaaaaccaaaaaacdaaaaaaceaaaaaacfaaaaaacgaaaaaachaaaaaaciaaaaaacjaaaaaackaaaaaaclaaaaaacmaaaaaacnaaaaaacoaaaaaacpaaaaaacqaaaaaacraaaaaacsaaaaaactaaaaaacuaaaaaacvaaaaaacwaaaaaacxaaaaaacyaaaaaaczaaaaaadbaaaaaadcaaaaaaddaaaaaadeaaaaaadfaaaaaadgaaaaaadhaaaaaadiaaaaaadjaaaaaadkaaaaaadlaaaaaadmaaaaaadnaaaaaadoaaaaaadpaaaaaadqaaaaaadraaaaaadsaaaaaadtaaaaaaduaaaaaadvaaaaaadwaaaaaadxaaaaaadyaaaaaadzaaaaaaebaaaaaaecaaaaaaedaaaaaaeeaaaaaaefaaaaaaegaaaaaaehaaaaaaeiaaaaaaejaaaaaaekaaaaaaelaaaaaaemaaaaaaenaaaaaaeoaaaaaaepaaaaaaeqaaaaaaeraaaaaaesaaaaaaetaaaaaaeuaaaaaaevaaaaaaewaaaaaaexaaaaaaeyaaaaaaezaaaaaafbaaaaaafcaaaaaafdaaaaaafeaaaaaaffaaaaaafgaaaaaafhaaaaaafiaaaaaafjaaaaaafkaaaaaaflaaaaaafmaaaaaafnaaaaaafoaaaaaafpaaaaaafqaaaaaafraaaaaafsaaaaaaftaaaaaafuaaaaaafvaaaaaafwaaaaaafxaaaaaafyaaaaaafzaaaaaagbaaaaaagcaaaaaagdaaaaaageaaaaaagfaaaaaaggaaaaaaghaaaaaagiaaaaaagjaaaaaagkaaaaaaglaaaaaagmaaaaaagnaaaaaagoaaaaaagpaaaaaagqaaaaaagraaaaaagsaaaaaagtaaaaaaguaaaaaagvaaaaaagwaaaaaagxaaaaaagyaaaaaagzaaaaaahbaaaaaahcaaaaaahdaaaaaaheaaaaaahfaaaaaahgaaaaaahhaaaaaahiaaaaaahjaaaaaahkaaaaaahlaaaaaahmaaaaaahnaaaaaahoaaaaaahpaaaaaahqaaaaaahraaaaaahsaaaaaahtaaaaaahuaaaaaahvaaaaaahwaaaaaahxaaaaaahyaaaaaahzaaaaaaibaaaaaaicaaaaaaidaaaaaaieaaaaaaifaaaaaaigaaaaaaihaaaaaaiiaaaaaaijaaaaaaikaaaaaailaaaaaaimaaaaaainaaaaaaioaaaaaaipaaaaaaiqaaaaaairaaaaaaisaaaaaaitaaaaaaiuaaaaaaivaaaaaaiwaaaaaaixaaaaaaiyaaaaaaizaaaaaajbaaaaaajcaaaaaajdaaaaaajeaaaaaajfaaaaaajgaaaaaajhaaaaaajiaaaaaajjaaaaaajkaaaaaajlaaaaaajmaaaaaajnaaaaaajoaaaaaajpaaaaaajqaaaaaajraaaaaajsaaaaaajtaaaaaajuaaaaaajvaaaaaajwaaaaaajxaaaaaajyaaaaaajzaaaaaakbaaaaaakcaaaaaakdaaaaaakeaaaaaakfaaaaaakgaaaaaakhaaaaaakiaaaaaakjaaaaaakkaaaaaaklaaaaaakmaaaaaaknaaaaaakoaaaaaakpaaaaaakqaaaaaakraaaaaaksaaaaaaktaaaaaakuaaaaaakvaaaaaakwaaaaaakxaaaaaakyaaaaaakzaaaaaalbaaaaaalcaaaaaaldaaaaaaleaaaaaalfaaaaaalgaaaaaalhaaaaaaliaaaaaaljaaaaaalkaaaaaallaaaaaalmaaaaaalnaaaaaaloaaaaaalpaaaaaalqaaaaaalraaaaaalsaaaaaaltaaaaaaluaaaaaalvaaaaaalwaaaaaalxaaaaaalyaaaaaalzaaaaaambaaaaaamcaaaaaamdaaaaaameaaaaaamfaaaaaamgaaaaaamhaaaaaamiaaaaaamjaaaaaamkaaaaaamlaaaaaammaaaaaamnaaaaaamoaaaaaampaaaaaamqaaaaaamraaaaaamsaaaaaamtaaaaaamuaaaaaamvaaaaaamwaaaaaamxaaaaaamyaaaaaamzaaaaaanbaaaaaancaaaaaandaaaaaaneaaaaaanfaaaaaangaaaaaanhaaaaaaniaaaaaanjaaaaaankaaaaaanlaaaaaanmaaaaaannaaaaaanoaaaaaanpaaaaaanqaaaaaanraaaaaansaaaaaantaaaaaanuaaaaaanvaaaaaanwaaaaaanxaaaaaanyaaaaaanzaaaaaaobaaaaaaocaaaaaaodaaaaaaoeaaaaaaofaaaaaaogaaaaaaohaaaaaaoiaaaaaaojaaaaaaokaaaaaaolaaaaaaomaaaaaaonaaaaaaooaaaaaaopaaaaaaoqaaaaaaoraaaaaaosaaaaaaotaaaaaaouaaaaaaovaaaaaaowaaaaaaoxaaaaaaoyaaaaaaozaaaaaapbaaaaaapcaaaaaapdaaaaaapeaaaaaapfaaaaaapgaaaaaaphaaaaaapiaaaaaapjaaaaaapkaaaaaaplaaaaaapmaaaaaapnaaaaaapoaaaaaappaaaaaapqaaaaaapraaaaaapsaaaaaaptaaaaaapuaaaaaapvaaaaaapwaaaaaapxaaaaaapyaaaaaapzaaaaaaqbaaaaaaqcaaaaaaqdaaaaaaqeaaaaaaqfaaaaaaqgaaaaaaqhaaaaaaqiaaaaaaqjaaaaaaqkaaaaaaqlaaaaaaqmaaaaaaqnaaaaaaqoaaaaaaqpaaaaaaqqaaaaaaqraaaaaaqsaaaaaaqtaaaaaaquaaaaaaqvaaaaaaqwaaaaaaqxaaaaaaqyaaaaaaqzaaaaaarbaaaaaarcaaaaaardaaaaaareaaaaaarfaaaaaargaaaaaarhaaaaaariaaaaaarjaaaaaarkaaaaaarlaaaaaarmaaaaaarnaaaaaaroaaaaaarpaaaaaarqaaaaaarraaaaaarsaaaaaartaaaaaaruaaaaaarvaaaaaarwaaaaaarxaaaaaaryaaaaaarzaaaaaasbaaaaaascaaaaaasdaaaaaaseaaaaaasfaaaaaasgaaaaaashaaaaaasiaaaaaasjaaaaaaskaaaaaaslaaaaaasmaaaaaasnaaaaaasoaaaaaaspaaaaaasqaaaaaasraaaaaassaaaaaastaaaaaasuaaaaaasvaaaaaaswaaaaaasxaaaaaasyaaaaaaszaaaaaatbaaaaaatcaaaaaatdaaaaaateaaaaaatfaaaaaatgaaaaaathaaaaaatiaaaaaatjaaaaaatkaaaaaatlaaaaaatmaaaaaatnaaaaaatoaaaaaatpaaaaaatqaaaaaatraaaaaatsaaaaaattaaaaaatuaaaaaatvaaaaaatwaaaaaatxaaaaaatyaaaaaatzaaaaaaubaaaaaaucaaaaaaudaaaaaaueaaaaaaufaaaaaaugaaaaaauhaaaaaauiaaaaaaujaaaaaaukaaaaaaulaaaaaaumaaaaaaunaaaaaauoaaaaaaupaaaaaauqaaaaaauraaaaaausaaaaaautaaaaaauuaaaaaauvaaaaaauwaaaaaauxaaaaaauyaaaaaauzaaaaaavbaaaaaavcaaaaaavdaaaaaaveaaaaaavfaaaaaavgaaaaaavhaaaaaaviaaaaaavjaaaaaavkaaaaaavlaaaaaavmaaaaaavnaaaaaavoaaaaaavpaaaaaavqaaaaaavraaaaaavsaaaaaavtaaaaaavuaaaaaavvaaaaaavwaaaaaavxaaaaaavyaaaaaavzaaaaaawbaaaaaawcaaaaaawdaaaaaaweaaaaaawfaaaaaawgaaaaaawhaaaaaawiaaaaaawjaaaaaawkaaaaaawlaaaaaawmaaaaaawnaaaaaawoaaaaaawpaaaaaawqaaaaaawraaaaaawsaaaaaawtaaaaaawuaaaaaawvaaaaaawwaaaaaawxaaaaaawyaaaaaawzaaaaaaxbaaaaaaxcaaaaaaxdaaaaaaxeaaaaaaxfaaaaaaxgaaaaaaxhaaaaaaxiaaaaaaxjaaaaaaxkaaaaaaxlaaaaaaxmaaaaaaxnaaaaaaxoaaaaaaxpaaaaaaxqaaaaaaxraaaaaaxsaaaaaaxtaaaaaaxuaaaaaaxvaaaaaaxwaaaaaaxxaaaaaaxyaaaaaaxzaaaaaaybaaaaaaycaaaaaaydaaaaaayeaaaaaayfaaaaaaygaaaaaayhaaaaaayiaaaaaayjaaaaaaykaaaaaaylaaaaaaymaaaaaaynaaaaaayoaaaaaaypaaaaaayqaaaaaayraaaaaaysaaaaaaytaaaaaayuaaaaaayvaaaaaaywaaaaaayxaaaaaayyaaaaaayzaaaaaazbaaaaaazcaaaaaazdaaaaaazeaaaaaazfaaaaaazgaaaaaazhaaaaaaziaaaaaazjaaaaaazkaaaaaazlaaaaaazmaaaaaaznaaaaaazoaaaaaazpaaaaaazqaaaaaazraaaaaazsaaaaaaztaaaaaazuaaaaaazvaaaaaazwaaaaaazxaaaaaazyaaaaaazzaaaaababaaaaabacaaaaabadaaaaabaeaaaaabafaaaaabagaaaaabahaaaaabaiaaaaabajaaaaabakaaaaabalaaaaabamaaaaabanaaaaabaoaaaaabapaaaaabaqaaaaabaraaaaabasaaaaabataaaaabauaaaaabavaaaaabawaaaaabaxaaaaabayaaaaabazaaaaabbbaaaaabbcaaaaabbdaaaaabbeaaaaabbfaaaaabbgaaaaabbhaaaaabbiaaaaabbjaaaaabbkaaaaabblaaaaabbmaaaaabbnaaaaabboaaaaabbpaaaaabbqaaaaabbraaaaabbsaaaaabbtaaaaabbuaaaaabbvaaaaabbwaaaaabbxaaaaabbyaaaaabbzaaaaabcbaaaaabccaaaaabcdaaaaabceaaaaabcfaaaaabcgaaaaabchaaaaabciaaaaabcjaaaaabckaaaaabclaaaaabcmaaaaabcnaaaaabcoaaaaabcpaaaaabcqaaaaabcraaaaabcsaaaaabctaaaaabcuaaaaabcvaaaaabcwaaaaabcxaaaaabcyaaaaabczaaaaabdbaaaaabdcaaaaabddaaaaabdeaaaaabdfaaaaabdgaaaaabdhaaaaabdiaaaaabdjaaaaabdkaaaaabdlaaaaabdmaaaaabdnaaaaabdoaaaaabdpaaaaabdqaaaaabdraaaaabdsaaaaabdtaaaaabduaaaaabdvaaaaabdwaaaaabdxaaaaabdyaaaaabdzaaaaabebaaaaabecaaaaabedaaaaabeeaaaaabefaaaaabegaaaaabehaaaaabeiaaaaabejaaaaabekaaaaabelaaaaabemaaaaabenaaaaabeoaaaaabepaaaaabeqaaaaaberaaaaabesaaaaabetaaaaabeuaaaaabevaaaaabewaaaaabexaaaaabeyaaaaabezaaaaabfbaaaaabfcaaaaabfdaaaaabfeaaaaabffaaaaabfgaaaaabfhaaaaabfiaaaaabfjaaaaabfkaaaaabflaaaaabfmaaaaabfnaaaaabfoaaaaabfpaaaaabfqaaaaabfraaaaabfsaaaaabftaaaaabfuaaaaabfvaaaaabfwaaaaabfxaaaaabfyaaaaabfzaaaaabgbaaaaabgcaaaaabgdaaaaabgeaaaaabgfaaaaabggaaaaabghaaaaabgiaaaaabgjaaaaabgkaaaaabglaaaaabgmaaaaabgnaaaaabgoaaaaabgpaaaaabgqaaaaabgraaaaabgsaaaaabgtaaaaabguaaaaabgvaaaaabgwaaaaabgxaaaaabgyaaaaabgzaaaaabhbaaaaabhcaaaaabhdaaaaabheaaaaabhfaaaaabhgaaaaabhhaaaaabhiaaaaabhjaaaaabhkaaaaabhlaaaaabhmaaaaabhnaaaaabhoaaaaabhpaaaaabhqaaaaabhraaaaabhsaaaaabhtaaaaabhuaaaaabhvaaaaabhwaaaaabhxaaaaabhyaaaaabhzaaaaabibaaaaabicaaaaabidaaaaabieaaaaabifaaaaabigaaaaabihaaaaabiiaaaaabijaaaaabikaaaaabilaaaaabimaaaaabinaaaaabioaaaaabipaaaaabiqaaaaabiraaaaabisaaaaabitaaaaabiuaaaaabivaaaaabiwaaaaabixaaaaabiyaaaaabizaaaaabjbaaaaabjcaaaaabjdaaaaabjeaaaaabjfaaaaabjgaaaaabjhaaaaabjiaaaaabjjaaaaabjkaaaaabjlaaaaabjmaaaaabjnaaaaabjoaaaaabjpaaaaabjqaaaaabjraaaaabjsaaaaabjtaaaaabjuaaaaabjvaaaaabjwaaaaabjxaaaaabjyaaaaabjzaaaaabkbaaaaabkcaaaaabkdaaaaabkeaaaaabkfaaaaabkgaaaaabkhaaaaabkiaaaaabkjaaaaabkkaaaaabklaaaaabkmaaaaabknaaaaabkoaaaaabkpaaaaabkqaaaaabkraaaaabksaaaaabktaaaaabkuaaaaabkvaaaaabkwaaaaabkxaaaaabkyaaaaabkzaaaaablbaaaaablcaaaaabldaaaaableaaaaablfaaaaablgaaaaablhaaaaabliaaaaabljaaaaablkaaaaabllaaaaablmaaaaablnaaaaabloaaaaablpaaaaablqaaaaablraaaaablsaaaaabltaaaaabluaaaaablvaaaaablwaaaaablxaaaaablyaaaaablzaaaaabmbaaaaabmcaaaaabmdaaaaabmeaaaaabmfaaaaabmgaaaaabmhaaaaabmiaaaaabmjaaaaabmkaaaaabmlaaaaabmmaaaaabmnaaaaabmoaaaaabmpaaaaabmqaaaaabmraaaaabmsaaaaabmtaaaaabmuaaaaabmvaaaaabmwaaaaabmxaaaaabmyaaaaabmzaaaaabnbaaaaabncaaaaabndaaaaabneaaaaabnfaaaaabngaaaaabnhaaaaabniaaaaabnjaaaaabnkaaaaabnlaaaaabnmaaaaabnnaaaaabnoaaaaabnpaaaaabnqaaaaabnraaaaabnsaaaaabntaaaaabnuaaaaabnvaaaaabnwaaaaabnxaaaaabnyaaaaabnzaaaaabobaaaaabocaaaaabodaaaaaboeaaaaabofaaaaabogaaaaabohaaaaaboiaaaaabojaaaaabokaaaaabolaaaaabomaaaaabonaaaaabooaaaaabopaaaaaboqaaaaaboraaaaabosaaaaabotaaaaabouaaaaabovaaaaabowaaaaazzzzz

Paste it via right click 3 times.

On the 3rd paste, you will receive the following:

[Picture](https://i.imgur.com/s7EWGNE.png)

Error for searching purposes

> Exception:
> System.ArgumentOutOfRangeException: The value must be greater than or equal to zero and less than the console's buffer size in that dimension.
> Parameter name: top
> Actual value was -29.
>    at System.Console.SetCursorPosition(Int32 left, Int32 top)
>    at Microsoft.PowerShell.PSConsoleReadLine.ReallyRender(RenderData renderData, String defaultColor)
>    at Microsoft.PowerShell.PSConsoleReadLine.ForceRender()
>    at Microsoft.PowerShell.PSConsoleReadLine.Insert(Char c)
>    at Microsoft.PowerShell.PSConsoleReadLine.SelfInsert(Nullable\`1 key, Object arg)
>    at Microsoft.PowerShell.PSConsoleReadLine.ProcessOneKey(ConsoleKeyInfo key, Dictionary`2 dispatchTable, Boolean ignoreIfNoAction, Object arg)
>    at Microsoft.PowerShell.PSConsoleReadLine.InputLoop()
>    at Microsoft.PowerShell.PSConsoleReadLine.ReadLine(Runspace runspace, EngineIntrinsics engineIntrinsics)

**Comment by msftrncs** [comment]
Please report the version of PSReadLine.   These issues are probably resolved in PSReadLine 2.0.0-beta5, available from PowerShellGallery.

```PowerShell
& {
    $hostName = $Host.Name
    if ($hostName -eq "ConsoleHost" -and (Get-Command Get-CimInstance -ErrorAction SilentlyContinue)) {
        $id = $PID
        $inWindowsTerminal = $false
        while ($true) {
            $p = Get-CimInstance -ClassName Win32_Process -Filter "ProcessId Like $id"
            if (!$p -or !$p.Name) { break }
            if ($p.Name -eq "WindowsTerminal.exe") { $inWindowsTerminal = $true; break }
            $id = $p.ParentProcessId
        }
        if ($inWindowsTerminal) { $hostName += " (Windows Terminal)" }
    }

    "`nPS version: $($PSVersionTable.PSVersion)"
    $v = (Get-Module PSReadline).Version
    $m = Get-Content "$(Split-Path -Parent (Get-Module PSReadLine).Path)\PSReadLine.psd1" | Select-String "Prerelease = '(.*)'"
    if ($m) {
        $v = "$v-" + $m.Matches[0].Groups[1].Value
    }
    "PSReadline version: $v"
    if ($IsLinux -or $IsMacOS) {
        "os: $(uname -a)"
    } else {
        "os: $((dir $env:SystemRoot\System32\cmd.exe).VersionInfo.FileVersion)"
    }
    "PS file version: $($name = if ($PSVersionTable.PSEdition -eq "Core") { "pwsh.dll" } else { "powershell.exe" }; (dir $pshome\$name).VersionInfo.FileVersion)"
    "HostName: $hostName"
    "BufferWidth: $([console]::BufferWidth)"
    "BufferHeight: $([console]::BufferHeight)`n"
}
```

**Comment by Reelix** [comment]
It seems that the default installation of Powershell did not have the PSReadLine Module installed (Not sure why it told me to go here then), although after upgrading to Powershell 7.0.0 Preview 4 (Which installed PSReadLine 2.0.0 to regular Powershell, and 2.0.0 beta 5 to Powershell 7.0.0 Preview 4) it seems that whilst the issue can be replicated in 2.0.0, it is indeed fixed in 2.0.0 beta 5.

**Comment by SteveL-MSFT** [comment]
Our plan is to update PSReadLine inbox in Windows once we get to 2.0.0-GA.  For now, please update to latest beta on PSGallery.
